### PR TITLE
Login form and view

### PR DESCRIPTION
**Pull request by subnomo**
Added the login view and a basic template.

### CODE REVIEW

File: app/views.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+    if request.method == 'POST':
+        username = request.form['username']
+        password = request.form['password']
+
+        # Login to TimeSync
+        ts = pymesync.TimeSync(baseurl="http://timesync-staging.osuosl.org/v1")
```
Comment (pop) [inline app/views.py]: this url should be in conf.py

**Comment by pop** [comment]
Looks good to me. A few pieces of feedback (unrelated to the files changed otherwise I would do them inline):
- `README.md` should include a line about 'how to run the app', as well as any setup to do before running in production.
- `conf.py` should really be `conf.py.dist` and the `README.md` will explain what to change before putting the app in production.
- This PR should be apart of the previous one. It's no big deal now, but we try to avoid merging failing tests into develop. It might be different for this project, but I don't remember hearing about us trying that.

The actual content of this PR looks good, there are just a few things to change (listed above).

**Comment by subnomo** [comment]
Some changes coming, so hold off on review.

**Comment by subnomo** [comment]
Ready for review.

**Comment by Kennric** [comment]
Looks pretty good so far. Please  create issues for your TODOs. +1

**Comment by pop** [comment]
This is looking pretty good. I would feel a bit better if travis was running our tests instead of just linting though.

**Comment by matthewrsj** [comment]
Tests pass and this is looking good, but :-1: until you run our tests with travis.

File: README.md [hunk 2]
```diff
@@ -9,6 +9,14 @@ $ source venv/bin/activate
 (venv) $ pip install -r requirements.txt
 ```
 
+To run the app, you should first edit the configuration:
+
+1. Rename the config file from `config.py.dist` to `config.py`.
+2. Change the `SECRET_KEY` to one of your own. Make it unique.
+4. Change `TIMESYNC_URL` to the URL of the TimeSync instance to
+communicate with.
+5. If you want to run it in debug mode, make sure you set `DEBUG` to `True`.
```
Comment (matthewrsj) [inline README.md:20]: You should also add the command to run the app:

```
python run.py
```

and explain that it is now running on `http://localhost:5000` and the login page can be accessed at `/login`.

File: app/views.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                                password=password,
+                                auth_type="password")
+
+        # TODO: Better error handling
+        if 'error' in token or 'pymesync error' in token:
+            print token
+            return 'There was an error.', 500
+
+        session['username'] = username
+        session['ts'] = token
```
Comment (matthewrsj) [inline app/views.py]: Is it going to be confusing having `ts` be a `pymesync.TimeSync` instance but `session['ts']` a JSON object containing `{'token': 'thereallylongtoken'}`?

You may also want to keep in mind that `session['ts']` does not just contain a token, but as explained above, a JSON object _containing_ a token. You will have to keep this in mind for re-auth in other views when you have to pass **just** a token to initiate pymesync.

The alternative to that is to set `session['ts']` (or `session['token']`) to `token['token']` so you don't have to worry about it later. You already verified there is no `error` in `token` so you shouldn't see any key errors. If you want to be even more explicit though you could do something like:

```
if 'token' in token:
    session['token'] = token['token']
else:
    print "unexpected response " + token
```

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+python:
+    - "2.7"
+
+notificaions:
+    email: false
+
+install:
+    - pip install -r requirements.txt
+
+script:
```
Comment (matthewrsj) [inline .travis.yml]: You are going to want to add something like this here (before this line):

```
before_script:
  - mv config.py.dist config.py
```

and additionally take the steps to set up the test environment as described in the README.

**Comment by subnomo** [comment]
@ElijahCaine @Kennric Review?

**Comment by matthewrsj** [comment]
@subnomo you'll need to rebase this. ping me if you need help.

**Comment by pop** [comment]
Tests++ are always good, the code looks good, everything looks good to me. +1 :+1:

**Comment by matthewrsj** [comment]
This is understandable, tests pass. I think this is ready to merge. +1

**Pull request merged**